### PR TITLE
Match function

### PR DESCRIPTION
**Pull request by pjebs**
https://github.com/gofiber/fiber/issues/2138

### CODE REVIEW

**Comment by pjebs** [comment]
There is one failing test (to do with escaping `:`).
Can someone help me diagnose it?

**Comment by li-jin-gou** [comment]
Please follow the PR template @pjebs

**Comment by efectn** [comment]
Can you add some benchmarks

**Comment by ReneWerner87** [comment]
- [ ] add some benchmarks

**Comment by pjebs** [comment]
What should it benchmark?
What input values should go into the function?

**Comment by efectn** [comment]
Can you check last comments @pjebs

**Comment by ReneWerner87** [comment]
@pjebs friendly reminder 
we need the documentation part and some benchmarks so we can optimize this functionality in the future

**Comment by pjebs** [comment]
It's still got a failed test case I need to investigate.

**Comment by pjebs** [comment]
This PR is now compete.
I don't think there is any need for documentation on the website. godoc documentation should be adequate.
Benchmark has been added.

File: path_test.go [hunk 1]
```diff
@@ -852,3 +1305,266 @@ func Benchmark_Path_matchParams(t *testing.B) {
...
+// 		match bool
+// 	}
+// 	testCase := func(pattern string, cases []testparams) {
+// 		for _, c := range cases {
+// 			match := RoutePatternMatch(c.url, pattern)
+// 			utils.AssertEqual(t, c.match, match, fmt.Sprintf("route: '%s', url: '%s'", pattern, c.url))
+// 		}
+// 	}
+// 	testCase("/api/v1/:param/*", []testparams{
+// 		{url: "/api/v1/entity", match: true},
```
Comment (efectn) [inline path_test.go:1323]: Are these tests commented?

Comment (pjebs) [reply path_test.go]: They were from above and used as a template. They've now been removed.

**Comment by ReneWerner87** [comment]
> 
all features we provide to the user are explained in our markdown documentation with examples and further information, why should we break with that for this feature ?

**Comment by pjebs** [comment]
@ReneWerner87 Where in docs should this function be documented? I don't see a good section for it.

**Comment by ReneWerner87** [comment]
> @ReneWerner87 Where in docs should this function be documented? I don't see a good section for it.

Good point
normally we would need another section which utils which are usable for all documented

since we don't have this and I am okay with your statement that the godocs are currently sufficient

**Comment by pjebs** [comment]
Can this be merged and tagged?

**Comment by efectn** [comment]
> Can this be merged and tagged?

Can you create a PR for docs repo? Route category may be nice for this feature.

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**